Packs/Mimecast/Integrations/MimecastEventCollector/MimecastEventCollector.py

```diff
@@ -0,0 +1,588 @@
+    """
+    if not audit_events:
+        return ''
+    else:
+        return audit_events[0].get('eventTime', '')
+
+
+def handle_last_run_exit(siem_event_handler: MimecastGetSiemEvents, audit_events: list):
+    """
+    This function removes duplicates from audit_events.
+    prepares the next dedup audit event list
+    sets the new demisto.LastRun
+
+    Args:
+        siem_event_handler (MimecastGetSiemEvents): the siem event handler.
+        audit_events (list): the audit events of this run.
+    Returns:
+        None
+    """
+    demisto_last_run = demisto.getLastRun()
+    # handle audit events last run
+    audit_events = dedup_audit_events(audit_events, demisto_last_run.get(AUDIT_EVENT_DEDUP_LIST, []))
+    audit_next_run = set_audit_next_run(audit_events) if set_audit_next_run(audit_events) else demisto_last_run.get(
+        AUDIT_LAST_RUN)
+    potential_duplicates_audit = prepare_potential_audit_duplicates_for_next_run(audit_events, audit_next_run)
+    audit_dedup_next_run = potential_duplicates_audit if potential_duplicates_audit else demisto_last_run.get(
+        AUDIT_EVENT_DEDUP_LIST)
+    # handle siem events last run
+    siem_next_run = siem_event_handler.token if siem_event_handler.token else demisto_last_run.get(SIEM_LAST_RUN)
+    siem_fetched_events_for_next_run = siem_event_handler.events_from_prev_run if \
+        siem_event_handler.events_from_prev_run else demisto_last_run.get(SIEM_EVENTS_FROM_LAST_RUN)
+
+    next_run_obj = {SIEM_LAST_RUN: siem_next_run,
+                    SIEM_EVENTS_FROM_LAST_RUN: siem_fetched_events_for_next_run,  # setting for next run
+                    AUDIT_LAST_RUN: audit_next_run,
+                    AUDIT_EVENT_DEDUP_LIST: audit_dedup_next_run}
+    demisto.setLastRun(next_run_obj)
+    demisto.info(f'\naudit events next run: {audit_next_run} \n siem next run: {siem_next_run} \n'
+                 f'audit potential dups: {audit_dedup_next_run}\n'
+                 f'siem_events_for_next_run: {len(siem_fetched_events_for_next_run)}\n')
+
+
+def prepare_potential_audit_duplicates_for_next_run(audit_events: list, next_run_time: str) -> list:
+    """
+    Notice: This function modifies the audit_events list
+    The list is sorted s.t. Latest events are in the start,
+    if no more events with the same time are found break the search.
+
+    Args:
+        audit_events (list): the list of the audit events
+        next_run_time (str): the new last_run type for next run
+
+    Return:
+        list: The event list for next run to check against duplicates.
+    """
+    if not audit_events or not next_run_time:
+        return []
+
+    same_time_events = []
+    for event in audit_events:
+        if event.get('eventTime', '') == next_run_time:
+            same_time_events.append(event.get('id'))
+        else:
+            break
+
+    return same_time_events
+
+
+def gather_events(siem_events: list, audit_events: list) -> list:
```

Comment:
why is that a function?
use `siem_events + audit_events`

Reply:
Done